### ISSUE TITLE
Support AST manipulation

### ISSUE THREAD

**Issue by kevinburke** [naoina/toml#36]
Say I wanted to load a TOML file from a program, change a value, and write it back to disk.

The current configuration makes that difficult. Say I wanted to change the value in `testdata/example.toml`. I would call `toml.Parse()` and eventually get an `ast.Value`, but there's no way to modify that, and updating the `Value` would not update the ast.Table.Data. In addition, it doesn't seem like there's a way to convert an `*ast.Table` back into a `[]byte`.

**Comment by fjl** [comment]
You're right, there is no built-in support for modifying the AST. It would be interesting to know what your use case is. 

You can do simple transformations by parsing into `interface{}`, modifying the result and finally writing it back out using Marshal. This is lossy because formatting and comments are lost. If the AST had support for modifications, how should comments/formatting be handled?

**Comment by kevinburke** [comment]
Use case is writing an Go program to manipulate values in a user's TOML
file, say to bump a version number, or change the value of a string. I
can't do that if it loses comments.


--
Kevin Burke
925.271.7005 | kev.inburke.com
